src/input/input_handler.cpp

```diff
@@ -0,0 +1,694 @@
+         i++) {
+        if (i[output_array] == parsed) {
+            return &output_array[i];
+        }
+    }
+    return &output_array[i];
+}
+
+void ParseInputConfig(const std::string game_id = "") {
+
+    const auto config_file = Config::GetFoolproofKbmConfigFile(game_id);
+
+    // todo: change usages of this to GetFoolproofKbmConfigFile (in the gui)
+    if (game_id == "") {
+        return;
+    }
+
+    // we reset these here so in case the user fucks up or doesn't include this,
+    // we can fall back to default
+    connections.clear();
+    mouse_deadzone_offset = 0.5;
+    mouse_speed = 1;
+    mouse_speed_offset = 0.125;
+    int lineCount = 0;
+
+    std::ifstream file(config_file);
+    std::string line = "";
+    while (std::getline(file, line)) {
+        lineCount++;
+        // strip the ; and whitespace
+        line.erase(std::remove(line.begin(), line.end(), ' '), line.end());
+        if (line[line.length() - 1] == ';') {
+            line = line.substr(0, line.length() - 1);
+        }
+        // Ignore comment lines
+        if (line.empty() || line[0] == '#') {
+            continue;
+        }
+        // Split the line by '='
+        std::size_t equal_pos = line.find('=');
+        if (equal_pos == std::string::npos) {
+            LOG_WARNING(Input, "Invalid format at line: {}, data: \"{}\", skipping line.",
+                        lineCount, line);
+            continue;
+        }
+
+        std::string output_string = line.substr(0, equal_pos);
+        std::string input_string = line.substr(equal_pos + 1);
+        std::size_t comma_pos = input_string.find(',');
+
+        if (output_string == "mouse_to_joystick") {
+            if (input_string == "left") {
+                mouse_joystick_binding = 1;
+            } else if (input_string == "right") {
+                mouse_joystick_binding = 2;
+            } else {
+                mouse_joystick_binding = 0; // default to 'none' or invalid
+            }
+            continue;
+        }
+        if (output_string == "key_toggle") {
+            if (comma_pos != std::string::npos) {
+                // handle key-to-key toggling (separate list?)
+                InputBinding toggle_keys = GetBindingFromString(input_string);
+                if (toggle_keys.KeyCount() != 2) {
+                    LOG_WARNING(Input,
+                                "Syntax error: Please provide exactly 2 keys: "
+                                "first is the toggler, the second is the key to toggle: {}",
+                                line);
+                    continue;
+                }
+                ControllerOutput* toggle_out =
+                    GetOutputPointer(ControllerOutput((OrbisPadButtonDataOffset)KEY_TOGGLE));
+                BindingConnection toggle_connection =
+                    BindingConnection(InputBinding(toggle_keys.key2), toggle_out, toggle_keys.key3);
+                connections.insert(connections.end(), toggle_connection);
```

Comment:
```suggestion
                connections.emplace_back(InputBinding(toggle_keys.key2), toggle_out, toggle_keys.key3);
```
Less lines and constructs object in-place. Ditto for all other cases of insert() like this